tests/autoyast/installation.pm

```diff
@@ -118,6 +118,12 @@ sub run {
             $num_errors++;
         }
         elsif (match_has_tag('warning-pop-up')) {
+            # See poo#19978, no timeout on partition warning, hence need to click OK button to soft-fail
+            if (check_screen('warning-partition-reduced', 0)) {
```

Comment:
you might misunderstand what match_has_tag does. If your needle has both warning-pop-up and warning-parition-reduced tags, assert_screen on 'warning-pop-up'. This sets $last_match - and you can then match_has_tag to check further tags on that match. No need to assert it